summarize_pr.py

```diff
@@ -42,29 +40,113 @@ def summarize_pr(
+        post_prompt="Analysis:",
+    )
+
+    latest_commit = api.pulls.list_commits(pr_number)[-1]["sha"]
+
+    # Create a comment on the PR
+    comment_content = f"""# Pull Request Summary from Gemini ✨\n{summarize_output}\n\n# Code Analysis from Gemini ✨\n{analysis_output}\n Generated at Commit: `{latest_commit}`"""
 
     # Post the summary as a comment
-    api.issues.create_comment(
-        pr_number, f"## Pull Request Summary from Gemini\n {response.text}"
+    api.issues.create_comment(pr_number, comment_content)
+
+    return summarize_output, analysis_output
+
+
+def convert_to_sarif(model_id: str, analysis_content: str) -> str:
+    model = GenerativeModel(
+        model_id,
+        system_instruction=[
+            "You are an expert software engineer.",
+        ],
     )
 
+    generation_config = GenerationConfig(
+        response_mime_type="application/json",
+        max_output_tokens=8192,
+    )
+
+    sarif_schema = requests.get(
+        "https://raw.githubusercontent.com/oasis-tcs/sarif-spec/123e95847b13fbdd4cbe2120fa5e33355d4a042b/Schemata/sarif-schema-2.1.0.json"
+    ).text
+
+    prompt = [
+        "The following is analysis of a GitHub Pull Request.",
+        "Your task is to convert this into SARIF (Static Analysis Results Interchange Format) format.",
+        "This is the SARIF schema:",
+        sarif_schema,
+        "Analysis Content:",
+        analysis_content,
+        "SARIF:"
+        """{"$schema": "https://json-schema.org/draft/2020-12/schema#", "version": "2.1.0", "runs": [{"tool": {"driver": {"name": "Gemini", "shortDescription": {"text": "A large language model capable of generating text, translating languages, writing different kinds of creative content, and answering your questions in an informative way."}, "fullDescription": {"text": "Gemini Pro is a large language model (LLM) developed by Google AI. It is trained on a massive dataset of text and code, and can be used for a wide variety of tasks, including text generation, translation, writing different kinds of creative content, and answering your questions in an informative way."}}}, "results":""",
+    ]
+
+    print("---Prompt---\n", prompt)
+    response = model.generate_content(prompt, generation_config=generation_config)
+    print("---Gemini Response---\n", response)
+
+    return response.text
+
+
+def upload_sarif(api: GhApi, pr_number: str, sarif_text: str):
+    encoded_data = base64.b64encode(gzip.compress(sarif_text.encode("utf-8"))).decode(
+        "utf-8"
+    )
+
+    commits = api.pulls.list_commits(pr_number)
+
+    sarif_response = api.code_scanning.upload_sarif(
+        commit_sha=commits[-1]["sha"],
+        sarif=encoded_data,
+        ref=f"refs/pull/{pr_number}/head",
+    )
+
+    return sarif_response
+
 
 if __name__ == "__main__":
     vertexai.init()
```

Comment:
The `upload_sarif` function effectively uploads the generated SARIF report to GitHub. 

It would be beneficial to include error handling to manage situations where the upload fails. This could involve implementing `try-except` blocks and logging errors for better debugging and user experience.